tests/base.py

```diff
@@ -0,0 +1,444 @@
+                    'updated_by',
+                },
+                self.PRIMARY_KEYS: {"id"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"date_updated"},
+                self.RESPECTS_START_DATE: True,
+            },
+
+            'tasks': {
+                self.AUTOMATIC_FIELDS: {
+                    'date_created',
+                    'date_updated',
+                    'id',
+                    'is_complete',
+                    'phone',
+                    'local_phone',
+                    'voicemail_url',
+                    'organization_id',
+                    '_type',
+                    'date',
+                    'voicemail_duration',
+                    'opportunity_value_currency',
+                    'contact_name',
+                    'body_preview',
+                    'phone_number_description',
+                    'text',
+                    'opportunity_note',
+                    'due_date',
+                    'updated_by',
+                    'opportunity_value',
+                    'email_id',
+                    'remote_phone',
+                    'created_by',
+                    'view',
+                    'object_type',
+                    'subject',
+                    'remote_phone_description',
+                    'contact_id',
+                    'emails',
+                    'lead_id',
+                    'recording_url',
+                    'remote_phone_formatted',
+                    'assigned_to',
+                    'is_dateless',
+                    'object_id',
+                    'opportunity_value_formatted',
+                    'updated_by_name',
+                    'phone_formatted',
+                    'created_by_name',
+                    'lead_name',
+                    'assigned_to_name',
+                    'opportunity_value_period',
+                },
+                self.PRIMARY_KEYS: {"id"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"date_updated"},
+                self.RESPECTS_START_DATE: True,
+            },
+
+            'users': {
+                self.AUTOMATIC_FIELDS: {
+                    'date_updated',
+                    'id',
+                    'last_name',
+                    'date_created',
+                    'organizations',
+                    'email',
+                    'first_name',
+                    'image',
+                    'last_used_timezone',
+                },
+                self.PRIMARY_KEYS: {"id"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"date_updated"},
+                self.RESPECTS_START_DATE: True,
+            }
+
+        }
+
+    def expected_automatic_fields(self):
+        auto_fields = {}
+        for k, v in self.expected_metadata().items():
+            auto_fields[k] = v.get(self.AUTOMATIC_FIELDS, set())
```

Comment:
This might just be that I'm unfamiliar with how we often write tests, but it's unintuitive that we have a dict key called `self.AUTOMATIC_FIELDS`, which sounds like it would be a list.